### DISCUSSION TITLE
Login Failed when user password contain '$'

### DISCUSSION THREAD

**Discussion by Rattnik** [apache/airflow#32508] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

When user creating through cli or LDAP Auth with password contain '**$**' (dollar symbol), unable to log in with error "**Login Failed for user**"

### How to reproduce

`airflow users create -e "test@test.com" -f "Firstname" -l "Lastname" -p "aaa$aaa" -u "test-user" -r Viewer`

### Operating System

"Debian GNU/Linux 11 (bullseye)"

### Versions of Apache Airflow Providers

apache-airflow==2.5.3

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by tirkarthi** [comment]
It looks like a case of $a being interpolated by shell within double quotes and not an Airflow issue. Can you please try using single quotes '' from the shell for -p?

**Reply by Rattnik** [reply]
Yes. It works. But which module uses to sync and create users from LDAP? Because it problem contain when I tried to first login with new user and password contain '$'

**Reply by potiuk** [reply]
Look at Flask App Builder documentation. This is where this part comes from

**Reply by potiuk** [reply]
(and if you find any errors - they should be reported there rather than here).

**Reply by potiuk** [reply]
https://flask-appbuilder.readthedocs.io/en/latest/

**Comment by potiuk** [comment]
Yes. You need to escape the characters that your shell expands.

**Comment by Rattnik** [comment]
> It looks like a case of $a being interpolated by shell within double quotes and not an Airflow issue. Can you please try using single quotes '' from the shell for -p?

**Comment by Nisogok** [comment]
Thank you for your work!
